### DISCUSSION TITLE
CI/CD fails with error Cannot read property *package json property* of undefined

### DISCUSSION THREAD

**Discussion by GuyNamedTom** [serverless/serverless#11773] [Q&A]
I'm having an issue with the CI/CD deployments. Everything was working fine until the other day when I started getting errors during the build step.  It seems that whatever the first package in the package.json dependencies will come back as "Cannot read property *package* of undefined."

`build step: npm ci --if-present --unsafe-perm

npm ERR! Cannot read property '@aws-sdk/credential-provider-node' of undefined`

I can deploy through the command line on my computer and it works just fine, it's just the CI/CD part that fails. There haven't been any new packages or plugins for Serverless since the last successful deploy. I'm using serverless-webpack if that makes any difference. Code for that is below.  Has anyone ran into this before? 

`const slsw = require('serverless-webpack');
const nodeExternals = require('webpack-node-externals');
const fs = require('fs');

console.log(slsw.lib.entries);
var nodeModules = {};
fs.readdirSync('node_modules')
  .filter(function (x) {
    return ['.bin'].indexOf(x) === -1;
  })
  .forEach(function (mod) {
    nodeModules[mod] = 'commonjs ' + mod;
  });

module.exports = {
  entry: slsw.lib.entries,
  externalsPresets: { node: true },
  mode: 'development',
  module: {
    rules: [
      {
        test: /\.js$/,
        exclude: /node_modules/,
        use: [
          {
            loader: 'babel-loader',
          },
        ],
      },
    ],
  },
  externals: nodeModules, // exclude external modules
};
`

The above is our staging environment. Now our development environment is running into another issue

`build step: npm ci --if-present --unsafe-perm

npm ERR! fsevents not accessible from jest-haste-map`

**Comment by robertsamarji** [comment]
I'm also getting this problem. Did you find a solution?

**Comment by NarHakobyan** [comment]
Same for me, any solution?

**Comment by GuyNamedTom** [comment]
@NarHakobyan  @robertsamarji 
I ended up having another person pushing up a commit to trigger the CI/CD, and it seemed to fix it. It was my account only that was having the issue. I've had the same issue locally pushing things up where I had to delete the .serverless file and it fixed the weird caching issue.

**Reply by Siemienik** [reply]
I suppose he/she has a different version of npm.

**Comment by robertsamarji** [comment]
Strange. My issue was I was using Node 18. Serverless support told me that their CI/CD feature doesn't yet support Node 18, but are currently working on it - should be ready in a few weeks.

Downgrading to Node 16 solved it for me.
